errors_test.go

```diff
@@ -0,0 +1,34 @@
+package fargo
+
+// MIT Licensed (see README.md) - Copyright (c) 2013 Hudl <@Hudl>
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestHTTPResponseStatusCode(t *testing.T) {
+	tests := []struct {
+		input   error
+		present bool
+		code    int
+	}{
+		{nil, false, 0},
+		{errors.New("other"), false, 0},
+		{&unsuccessfulHTTPResponse{404, "missing"}, true, 404},
+	}
+	for _, test := range tests {
+		code, present := HTTPResponseStatusCode(test.input)
+		if present {
+			if !test.present {
+				t.Errorf("input %v: want absent, got code %d", test.input, code)
```

Comment:
Can you please do those tests using Convey. (see other test files for guideline)